lib/matplotlib/colors.py

```diff
@@ -1225,6 +1254,875 @@ def reversed(self, name=None):
         return new_cmap
 
 
+class MultivarColormap:
+    """
+    Class for holding multiple `~matplotlib.colors.Colormap` for use in a
+    `~matplotlib.cm.VectorMappable` object
+    """
+    def __init__(self, colormaps, combination_mode, name='multivariate colormap'):
+        """
+        Parameters
+        ----------
+        colormaps: list or tuple of `~matplotlib.colors.Colormap` objects
+            The individual colormaps that are combined
+        combination_mode: str, 'sRGB_add' or 'sRGB_sub'
+            Describe how colormaps are combined in sRGB space
+
+            - If 'sRGB_add' -> Mixing produces brighter colors
+              `sRGB = cmap[0][X[0]] + cmap[1][x[1]] + ... + cmap[n-1][x[n-1]]`
+            - If 'sRGB_sub' -> Mixing produces darker colors
+              `sRGB = cmap[0][X[0]] + cmap[1][x[1]] + ... + cmap[n-1][x[n-1]] - n + 1`
```

Comment:
I think writing this as `1 - (sum (1 - xi))` would be much clearer as it took me a bit of effort to backout why it was this way!

Reply:
@tacaswell does this work for you?
```py
"""
            - If 'sRGB_add' -> Mixing produces brighter colors
              `sRGB = np.sum([cm(X[i]) for i, cm in enumerate(cmaps)], axis = 0)`
            - If 'sRGB_sub' -> Mixing produces darker colors
              `sRGB = 1 - np.sum([1 - cm(X[i]) for i, cm in enumerate(cmaps)], axis = 0)`
"""
```
In my mind the expression we write should be valid python.
But we could do a simpler:
```py
"""
            - If 'sRGB_add' -> Mixing produces brighter colors
              `sRGB = sum(colors)`
            - If 'sRGB_sub' -> Mixing produces darker colors
              `sRGB = 1 - sum(1 - colors)`
"""
```

Reply:
@story645 can I get your preference on this?